src/twisted/web/test/test_flatten.py

```diff
@@ -269,14 +269,17 @@ def test_commentEscaping(self) -> Deferred[List[bytes]]:
         The data in a L{Comment} is escaped and mangled in the flattened output
         so that the result is a legal SGML and XML comment.
 
-        SGML comment syntax is complicated and hard to use. This rule is more
-        restrictive, and more compatible:
+        BNF (Backus-Naur Form) production rule for comments:
+        Comment	 ::=  '<!--' ((Char - '-') | ('-' (Char - '-')))* '-->'
```

Comment:
Just a suggestion, I think it would be helpful to present the production rule as code, rather than flowed in paragraph text:

```suggestion
        BNF (Backus-Naur Form) production rule for comments::

            Comment	 ::=  '<!--' ((Char - '-') | ('-' (Char - '-')))* '-->'
```